apps/web/src/utils/skillUtils.ts

```diff
@@ -330,11 +330,31 @@ export const sortPoolSkillsBySkillCategory = <T extends PoolSkill[]>(
  * @returns Skill[]
  */
 export const filterPoolSkillsByType = (
-  poolSkills: PoolSkill[],
+  poolSkills: Maybe<Maybe<PoolSkill>[]> | undefined,
   poolSkillType: PoolSkillType,
 ): Skill[] => {
-  const skills = poolSkills
+  const skills = unpackMaybes(poolSkills)
     .filter((poolSkill) => poolSkill.type === poolSkillType)
     .map((poolSkill) => poolSkill.skill);
   return unpackMaybes(skills);
 };
+
+export function groupPoolSkillByType(
+  poolSkills?: Maybe<Maybe<PoolSkill>[]>,
+): Record<PoolSkillType, Maybe<Array<Skill> | undefined>> {
+  const nonEmptyPoolSkills = unpackMaybes(poolSkills);
+  return {
+    [PoolSkillType.Essential]: filterPoolSkillsByType(
+      nonEmptyPoolSkills,
+      PoolSkillType.Essential,
+    ),
+    [PoolSkillType.Nonessential]: filterPoolSkillsByType(
+      nonEmptyPoolSkills,
+      PoolSkillType.Nonessential,
+    ),
+  };
+}
```

Comment:
What do you think about this alternative? Benefits:
- if the number of skill types ever expanded (unlikely, I know) this would still work
- subjective, but I've come to prefer Map objects when doing this kind of grouping. I think it helps indicate the purpose of the return value. I know this would require some minor re-writes elsewhere.
- this filters out any Maybes and undefineds
```suggestion
export function groupPoolSkillByType(
  poolSkills?: Maybe<Maybe<PoolSkill>[]>,
): Map<PoolSkillType, Array<Skill>> {
  return unpackMaybes(poolSkills).reduce((map, poolSkill) => {
    const { type, skill } = poolSkill;
    if (type && skill) {
      if (!map.has(type)) {
        map.set(type, []);
      }
      map.get(type)?.push(skill);
    }
    return map;
  }, new Map<PoolSkillType, Array<Skill>>());
}
```